packages/blocks/src/api/utils.js

```diff
@@ -184,53 +189,78 @@ export function getAccessibleBlockLabel(
 			return sprintf(
-				/* translators: accessibility text. 1: The block title. 2: The block row number. 3: The block label.. */
-				__( '%1$s Block. Row %2$d. %3$s' ),
-				title,
+				/* translators: accessibility text. 1: The block row number. 2: The total number of blocks. 3: The block label.. */
+				__( '%1$d of %2$d. %3$s' ),
 				position,
+				total,
+				label
+			);
+		} else if ( hasLabel && hasParentTitle ) {
+			return sprintf(
+				/* translators: accessibility text. 1: The block row number. 2: The total number of blocks. 3: The parent block title. 4: The block label.. */
+				__( '%1$d of %2$d. Child of %3$s block. %4$s' ),
+				position,
+				total,
+				parentTitle,
 				label
 			);
 		}
 
+		if ( hasParentTitle ) {
+			return sprintf(
+				/* translators: accessibility text. 1: The block row number. 2: The total number of blocks. 3: The parent block title. */
+				__( '%1$d of %2$d. Child of %3$s block.' ),
+				position,
+				total,
+				parentTitle
+			);
+		}
 		return sprintf(
-			/* translators: accessibility text. 1: The block title. 2: The block row number. */
-			__( '%1$s Block. Row %2$d' ),
-			title,
-			position
+			/* translators: accessibility text. 1: The block row number. 2: The total number of blocks. */
+			__( '%1$d of %2$d.' ),
+			position,
+			total
 		);
 	} else if ( hasPosition && direction === 'horizontal' ) {
 		if ( hasLabel ) {
 			return sprintf(
-				/* translators: accessibility text. 1: The block title. 2: The block column number. 3: The block label.. */
-				__( '%1$s Block. Column %2$d. %3$s' ),
-				title,
+				/* translators: accessibility text. 1: The block column number. 2: The total number of blocks. 3: The block label.. */
+				__( 'Column %1$d of %2$d. %4$s' ),
+				position,
+				total,
+				label
+			);
+		} else if ( hasLabel && hasParentTitle ) {
+			return sprintf(
+				/* translators: accessibility text. 1: The block column number. 2: The total number of blocks. 3: The block parent title. 4: The block label.. */
+				__( 'Column %1$d of %2$d. Child of %3$s block. %4$s' ),
 				position,
+				total,
+				parentTitle,
 				label
 			);
 		}
 
+		if ( hasParentTitle ) {
+			return sprintf(
+				/* translators: accessibility text. 1: The block column number. 2: The total number of blocks. 3: The block parent title. */
+				__( 'Column %1$d of %2$d. Child of %3$s block.' ),
+				position,
+				total,
+				parentTitle
+			);
+		}
 		return sprintf(
-			/* translators: accessibility text. 1: The block title. 2: The block column number. */
-			__( '%1$s Block. Column %2$d' ),
-			title,
-			position
+			/* translators: accessibility text. 1: The block column number. 2: The total number of blocks. */
+			__( 'Column %1$d of %2$d.' ),
+			position,
+			total
 		);
 	}
 
 	if ( hasLabel ) {
-		return sprintf(
-			/* translators: accessibility text. %1: The block title. %2: The block label. */
-			__( '%1$s Block. %2$s' ),
-			title,
-			label
-		);
+		return sprintf( '%1$s', label );
```

Comment:
Seems like  there's no need for the sprintf anymore.

Reply:
Implemented.